### PR TITLE
lbank rsa signing in py and php

### PR DESCRIPTION
**Pull request by frosty00**
don't merge until we fix js...

### CODE REVIEW

File: js/lbank.js [hunk 1]
```diff
@@ -501,8 +501,22 @@ module.exports = class lbank extends Exchange {
             let query = this.keysort (this.extend ({
                 'api_key': this.apiKey,
             }, params));
-            let queryString = this.rawencode (query) + '&secret_key=' + this.secret;
-            query['sign'] = this.hash (this.encode (queryString)).toUpperCase ();
+            let queryString = this.rawencode (query);
+            let message = this.hash (this.encode (queryString)).toUpperCase ();
+            let secretArr = ['-----BEGIN RSA PRIVATE KEY-----'];
+            let secretLength = true ? this.secret.length : null;  // eslint-disable-line
```
Comment (frosty00) [inline js/lbank.js]: let me know if there is a more standard way to get `strlen` in php.

Comment (kroitor) [reply js/lbank.js]: Yes, there is (the above line won't work) https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#working-with-array-lengths

Comment (kroitor) [reply js/lbank.js]: Pardon, ignore my previous comment, your code is fine.

File: js/lbank.js [hunk 2]
```diff
@@ -501,8 +501,22 @@ module.exports = class lbank extends Exchange {
...
+            for (let i = 0; i < secretLength; i++) {
+                let start = i * 64;
+                let end = start + 64;
```
Comment (kroitor) [inline js/lbank.js]: This won't work, though: https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#adding-numbers-and-concatenating-strings

It has to be: `let end = this.sum (start, 64);`

**Comment by frosty00** [comment]
fyi node-rsa distinguishes between pkcs8 and pkcs1 by using 

```
-----BEGIN PRIVATE KEY-----
```

vs

```
-----BEGIN RSA PRIVATE KEY-----
```

py/php are able to autodetect based on the body bytes.

File: js/base/functions/crypto.js [hunk 3]
```diff
@@ -27,35 +27,42 @@ const hmac = (request, secret, hash = 'sha256', digest = 'hex') => {
...
     const token = [ encodedHeader, encodedData ].join ('.')
-    if (!(alg in algos)) {
-        throw new ExchangeError (alg + ' is not a supported jwt algorithm.')
-    }
     const algoType = alg.slice (0, 2);
     const algorithm = algos[alg]
     let signature = undefined
     if (algoType === 'HS') {
-        signature = urlencodeBase64 (utf16ToBase64 (hmac (token, secret, algorithm, 'utf16')))
+        signature = urlencodeBase64 (hmac (token, secret, algorithm, 'base64'))
```
Comment (kroitor) [inline js/base/functions/crypto.js]: The problem with it is that if it was done this way, then it probably had to be done so for some other exchange...

Comment (kroitor) [reply js/base/functions/crypto.js]: https://github.com/ccxt/ccxt/commit/8df40f78562502b545adf66e5ebc2c48187a4e5d

Comment (kroitor) [reply js/base/functions/crypto.js]: ... and oceanex.

**Pull request merged**